src/Symfony/Component/Yaml/Inline.php

```diff
@@ -592,6 +592,8 @@ private static function evaluateScalar(string $scalar, int $flags, array $refere
                             $i = 0;
                             if (\defined($const = self::parseScalar(substr($scalar, 11), 0, null, $i, false))) {
                                 return \constant($const);
+                            } elseif ('::class' === strtolower(substr($const, -7)) && class_exists($class = (string) substr($const, 0, -7))) {
```

Comment:
this is broken for several reasons:

- it ignores interfaces (and traits)
- PHP does not require a class name to exist to resolve `::class` (which is not an actual constant)

Reply:
I was thinking about validation, adding this would permit to ensure that the class really exists.
If the class disappears later, the YAML parser will throw an exception.

Reply:
but that's **not** the behavior of `::class` in PHP.

Reply:
Yes I know, I read the documentation before ;)
`:class` in PHP will always return the class before it, without validating it

Reply:
and, and that's precisely why a YAML tag dealing with PHP constants should not add a validation which does not exist in PHP.

Reply:
By the way, I submitted it to discuss this topic, as I was doing some refactoring where some classes were referenced as string but were not anymore present, adding this feature could detect that they are no longer existing and avoid some runtime exceptions later.

Reply:
that's not something named `!php/const` should be doing, as what you ask is precisely not the behavior of the PHP `::class` constant.

Reply:
Ok, let's close then. That's was also controversial on my side, I will add custom tag on my code to handle that